syntax = "proto3";
package population.proto;

import "market/proto/goods.proto";

message Consumption {
  market.proto.Container consumed = 1;
  market.proto.Container capital = 2;
}

message ConsumptionPackage {
  Consumption food = 1;
  Consumption shelter = 2;
  Consumption culture = 3;

  // Empty indicates the package is usable by all cultures.
  map<string, bool> allowed_cultures = 4;
}

message ConsumptionLevel {
  repeated ConsumptionPackage packages = 1;
}

message PopUnit {
  uint64 pop_id = 1;
  repeated int32 males = 2;
  repeated int32 women = 3;
  market.proto.Container wealth = 4;
  string culture = 5;
}

